### ISSUE TITLE
make remove_bumps and normalize return arrays, not change in-place

### ISSUE THREAD

**Issue by gmgeorg** [tpospisi/DeepCDE#6]
code would work more nicely with concepts of layers (inputs and outputs), if remove_bumps and normalize dont change in place, but just work like layers: accept inputs and return modified outputs.  Also helps in interactive use of these functions to not accidentally change results from `cde_predict` when calling normalize on output again

Could even go as far as making `normalize` a keras/pytorch layer that can be added as part of the network architectures as a penalty / regularization.
